cmake/faceliftMacros.cmake

```diff
@@ -402,13 +428,77 @@ endmacro()
 
 macro(_facelift_add_target_start IMPLEMENTATION_TARGET_NAME)
 
-    set(HEADERS_TO_BE_MOCCED ${HEADERS} ${HEADERS_NO_INSTALL})
+    set(CURRENT_GENERATED_DIR  ${CMAKE_BINARY_DIR}/facelift_generated/${IMPLEMENTATION_TARGET_NAME})
+
+    # exclude some auto-generated files from moc processing
+    set(HEADERS_NO_MOC_WILDCARD
+       ${CURRENT_GENERATED_DIR}/*/Module.h
+       ${CURRENT_GENERATED_DIR}/*/ModuleIPC.h
+       ${CURRENT_GENERATED_DIR}/*IPCCommon.h
+    )
+
+    foreach(HEADER_NO_MOC_WILDCARD ${HEADERS_NO_MOC_WILDCARD})
+        file(GLOB_RECURSE GLOB_FILES ${HEADER_NO_MOC_WILDCARD})
+        list(APPEND HEADERS_NO_MOC_GLOB ${GLOB_FILES})
+    endforeach()
+
+    # but some files can be as processed as not by the moc
+    # we will analyse their text for special macro later (MOC_MACRO_NAMES)
+    set(HEADERS_CAN_MOC_WILDCARD
+       ${CURRENT_GENERATED_DIR}/*ModulePrivate.h
+    )
+
+    set(MOC_MACRO_NAMES "Q_OBJECT" "Q_GADGET" "Q_NAMESPACE")
+
+    unset(HEADERS_CAN_MOC)
+    foreach(HEADER ${HEADERS_CAN_MOC_WILDCARD})
+        file(GLOB_RECURSE GLOB_FILES ${HEADER})
+        list(APPEND HEADERS_CAN_MOC ${GLOB_FILES})
+    endforeach()
+
+    # verify if files that can be processed by moc contains MOC_MACRO_NAMES
+    if(HEADERS_CAN_MOC)
+        foreach(HEADER_CAN_HAVE_MOC ${HEADERS_CAN_MOC})
+            file(READ ${HEADER_CAN_HAVE_MOC} FILE_STRING)
+
+            unset(IS_FOUND)
+            foreach(MACRO_NAME ${MOC_MACRO_NAMES})
+                string(FIND "${FILE_STRING}" "${MACRO_NAME}" IS_FOUND)
+
+                if(${IS_FOUND} GREATER_EQUAL 0)
+                    break()
+                endif()
+
+            endforeach()
+
+            if(${IS_FOUND} LESS 0)
+                list(APPEND HEADERS_NO_MOC_GLOB ${HEADER_CAN_HAVE_MOC})
+            endif()
+
+        endforeach()
+    endif()
+
+    # we prepend string like ";header1.h;header2.h;" and then replace 
+    # like ;header1.h just not to replace header like extraheader1.h
```

Comment:
I think this comment is not relevant anymore

Reply:
you are right !